oak_io/src/serde.rs

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Implementations of `serde`'s `Deserialize` and `Serialize` traits for [`Sender`] and
+//! [`Receiver`].
+//!
+//! Both types serialize to the unit struct `()`. Combine `serde` serialization with the
+//! [`HandleVisit`] trait to serialize data containing handles.
+
+use crate::{handle::HandleVisit, Decodable, Encodable, Receiver, Sender};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+impl<T: Encodable + Decodable + HandleVisit> Serialize for Receiver<T> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_unit()
+    }
+}
+
+impl<'de, T: Encodable + Decodable + HandleVisit> Deserialize<'de> for Receiver<T> {
+    fn deserialize<D: Deserializer<'de>>(_: D) -> Result<Self, D::Error> {
+        Ok(Receiver::default())
+    }
+}
+
+impl<T: Encodable + Decodable + HandleVisit> Serialize for Sender<T> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_unit()
+    }
+}
+
+impl<'de, T: Encodable + Decodable + HandleVisit> Deserialize<'de> for Sender<T> {
+    fn deserialize<D: Deserializer<'de>>(_: D) -> Result<Self, D::Error> {
+        Ok(Sender::default())
+    }
```

Comment:
This would benefit from some extra documentation, in particular clarifying that they are not meant to be serialized by themselves, but rather via the encode and decode methods, and a bit of a high level description (and perhaps some simple test cases) to show what happens if they are serialized with serde (e.g. handle numbers are lost on the round trip conversion).

Reply:
I have tried to make this clear in the top-level module comment, would you prefer to have it on the `impl`s themselves?

I would link to `serde_encoding` here, but I don't think we allow for cross-crate documentation links? 
Adding test cases seems a bit overkill to me, especially since we would need to add a crate like `bincode` to the test dependencies to write those tests.